src/components/ModalComponent.tsx

```diff
@@ -1,69 +1,51 @@
-import React, { useState } from 'react';
+import React, { MouseEvent } from 'react';
 import { withRouter, RouteComponentProps } from 'react-router';
-import { publishQuestion } from '../services/api-service';
 
 interface IProps {
-  text: string;
-  id?: string;
-  isDelete?: boolean;
-  isPublish?: boolean;
-  isModalOpen(value: boolean);
+  content: string;
+  successButtonText?: string;
+  warningButtonText?: string;
+  successCallback?(e: MouseEvent): void;
+  warningCallback?(e: MouseEvent): void;
+  handleClose(): void;
+  hideButtons?: boolean;
 }
 
 const ModalComponent = (props: IProps & RouteComponentProps) => {
-  const { isPublish, isDelete, isModalOpen, text, id, history } = props;
-  const [modalText, setModalText] = useState<string>(text);
-  const [isVisible, setIsVisible] = useState<boolean | undefined>(
-    isPublish || isDelete,
-  );
-
-  const onPublishQuestion = event => {
-    if (!id) return;
-    publishQuestion(id)
-      .then(() => {
-        setModalText('Svaret er nå publisert!');
-        setIsVisible(false);
-      })
-      .catch(() => {
-        setIsVisible(false);
-        setModalText('Noe gikk galt.');
-      });
-    setTimeout(() => history.goBack(), 3000);
-    event.preventDefault();
-  };
-
-  const onAbort = event => {
-    setIsVisible(false);
-    setModalText(
-      'Svaret er sendt til eleven, men ble ikke publisert på Digitalleksehjelp.no',
-    );
-    setTimeout(() => history.goBack(), 3000);
-    event.preventDefault();
-  };
+  const {
+    content,
+    successButtonText,
+    warningButtonText,
+    successCallback,
+    warningCallback,
+    handleClose,
+    hideButtons,
+  } = props;
 
   return (
-    <div className={`modal`}>
-      <p>{modalText}</p>
-      <button
-        className="leksehjelp--button-close"
-        onClick={() => isModalOpen(false)}
-      >
+    <div className="modal">
+      <p>{content}</p>
+      <button className="leksehjelp--button-close" onClick={handleClose}>
         x
```

Comment:
Vi bør gjøre noe med denne... 🙈 
Meeen om vi gjør det med `.css` eller `.svg` vet jeg ikke? @sirius har du noe anbefalinger?

Reply:
Hva mener du med å gjøre noe med den? Antar du mener styling?

Reply:
Nei, vi bør ha noe annet en krysset. Det er meg som la det opp der. Men det ser ikke bra ut :)

Reply:
Okei, trenger vi noe i det hele tatt? å trykke escape f.eks. er ganske universelt for å exite  modaler